SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/commands/IntegrityCheck.swift

```diff
@@ -0,0 +1,56 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import ArgumentParser
+import Foundation
+
+struct IntegrityCheck: ParsableCommand {
+  static var configuration: CommandConfiguration {
+    return CommandConfiguration(
+      abstract: """
+        Ensure changes are fully tested, formatted, and validated before pull request submission.
+        """
+    )
+  }
+
+  @Flag(help: "Enable verbose logging.")
+  var verbose: Bool = false
+
+  func run() throws {
+    try runCommand("format")
+
+    try runCommand("generate-source-code")
+
+    try runCommand("test", "--use-remote-deps")
+
+    try runCommand("build", "--use-remote-deps")
```

Comment:
I'm uncertain whether we need to build everything considering that we are executing all the tests. Theoretically, `swift-parser-cli` and `Editor Extension` lack tests, so we don't include them in the previously mentioned command. However, is it essential to run them as an 'integrity check' before committing changes?

Reply:
I would build before running test. 
That's the same as CI